### PR TITLE
chores: add poller unit tests

### PR DESCRIPTION
**Pull request by stephhuynh18**

### CODE REVIEW

File: cmd/cas/main.go [hunk 1]
```diff
@@ -93,6 +94,6 @@ func main() {
 	wg := sync.WaitGroup{}
 	wg.Add(1)
 	// Poll from Anchor DB and post to the Validate queue to kick-off processing
-	go services.NewRequestPoller(anchorDb, stateDb, validateQueue).Run()
+	go services.NewRequestPoller(anchorDb, stateDb, validateQueue).Run(context.Background())
```
Comment (smrz2001) [inline cmd/cas/main.go:97]: This made me remember we need to add graceful shutdown sometime soon (it's already one of the TODOs in the readme).

We should be able to send a shutdown signal and have all services gracefully cleanup and terminate. Right now, we have to kill the task.

File: services/poll_test.go [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+// if no requests do not publish anything and try again after a second
+
+func TestPoller(t *testing.T) {
+	tests := map[string]struct {
+		readyPublisher            *FakeReadyPublisher
+		expectedCheckpoints       []time.Time
+		expectedCurrentCheckpoint time.Time
+	}{
+		"Can poll": {
```
Comment (smrz2001) [inline services/poll_test.go:82]: Would you mind terribly adding some comments for the timeline each test is checking for 😅?

My ADHD brain just can't follow, though I see what you're testing for.

e.g.
- T0s: Test starts
- T60s: First message sent
and so on

**Review by smrz2001 — APPROVED** [review]
Looks great! Just a minor request for some additional comments.

File: services/poll_test.go (hunk 2 above)
Comment (stephhuynh18) [reply services/poll_test.go:82]: done! Channels can quite hard to follow (the libp2p library makes more sense now)

Comment (smrz2001) [reply services/poll_test.go:82]: Ty, looks good!
